### ISSUE TITLE
Windows 10 - Extraterm always opens new console windows instead of new tabs

### ISSUE THREAD

**Issue by ctoma** [sedwards2009/extraterm#230]
Hi everyone,

Once I install Extraterm, every shell opens in a new window like this https://i.imgur.com/LYm3ogd.png
It doesn't matter if it's windows cmd, powershell, git bash, cmder - I cannot figure out how to set it up to open within the Extraterm window. There is a cursor in the Extraterm window but it does not accept any commands. Right now all Extraterm seems to do is launch another window and nothing more.

Is it supposed to work natively out of the box? Is there something basic I am missing here? I've read all the documentation and searched github issues but can't see any mention of my issue.

I'm not sure if it's relevant or not but here are the developer errors:

`Uncaught Error: spawn wsl.exe ENOENT
    at Process.ChildProcess._handle.onexit (internal/child_process.js:246)
    at onErrorNT (internal/child_process.js:429)
    at processTicksAndRejections (internal/process/task_queues.js:81)
C:\Program Files\Extraterm\resources\app\extraterm\node_modules\source-map-support\source-map-support.js:80 [Deprecation] Synchronous XMLHttpRequest on the main thread is deprecated because of its detrimental effects to the end user's experience. For more help, check https://xhr.spec.whatwg.org/.
retrieveSourceMapURL @ C:\Program Files\Extraterm\resources\app\extraterm\node_modules\source-map-support\source-map-support.js:80
retrieveSourceMap @ C:\Program Files\Extraterm\resources\app\extraterm\node_modules\source-map-support\source-map-support.js:110
mapSourcePosition @ C:\Program Files\Extraterm\resources\app\extraterm\node_modules\source-map-support\source-map-support.js:140
wrapCallSite @ C:\Program Files\Extraterm\resources\app\extraterm\node_modules\source-map-support\source-map-support.js:303
(anonymous) @ C:\Program Files\Extraterm\resources\app\extraterm\node_modules\source-map-support\source-map-support.js:338
prepareStackTrace @ C:\Program Files\Extraterm\resources\app\extraterm\node_modules\source-map-support\source-map-support.js:337
C:\Program Files\Extraterm\resources\app\extraterm\node_modules\source-map-support\source-map-support.js:81 GET file:///C:/Program%20Files/Extraterm/resources/app/extraterm/src/render_process/internal/child_process.js net::ERR_FILE_NOT_FOUND
retrieveSourceMapURL @ C:\Program Files\Extraterm\resources\app\extraterm\node_modules\source-map-support\source-map-support.js:81
retrieveSourceMap @ C:\Program Files\Extraterm\resources\app\extraterm\node_modules\source-map-support\source-map-support.js:110
mapSourcePosition @ C:\Program Files\Extraterm\resources\app\extraterm\node_modules\source-map-support\source-map-support.js:140
wrapCallSite @ C:\Program Files\Extraterm\resources\app\extraterm\node_modules\source-map-support\source-map-support.js:303
(anonymous) @ C:\Program Files\Extraterm\resources\app\extraterm\node_modules\source-map-support\source-map-support.js:338
prepareStackTrace @ C:\Program Files\Extraterm\resources\app\extraterm\node_modules\source-map-support\source-map-support.js:337
C:\Program Files\Extraterm\resources\app\extraterm\node_modules\source-map-support\source-map-support.js:47 GET file:///C:/Program%20Files/Extraterm/resources/app/extraterm/src/render_process/internal/child_process.js net::ERR_FILE_NOT_FOUND
retrieveFile @ C:\Program Files\Extraterm\resources\app\extraterm\node_modules\source-map-support\source-map-support.js:47
retrieveSourceMapURL @ C:\Program Files\Extraterm\resources\app\extraterm\node_modules\source-map-support\source-map-support.js:93
retrieveSourceMap @ C:\Program Files\Extraterm\resources\app\extraterm\node_modules\source-map-support\source-map-support.js:110
mapSourcePosition @ C:\Program Files\Extraterm\resources\app\extraterm\node_modules\source-map-support\source-map-support.js:140
wrapCallSite @ C:\Program Files\Extraterm\resources\app\extraterm\node_modules\source-map-support\source-map-support.js:303
(anonymous) @ C:\Program Files\Extraterm\resources\app\extraterm\node_modules\source-map-support\source-map-support.js:338
prepareStackTrace @ C:\Program Files\Extraterm\resources\app\extraterm\node_modules\source-map-support\source-map-support.js:337
C:\Program Files\Extraterm\resources\app\extraterm\node_modules\source-map-support\source-map-support.js:81 GET file:///C:/Program%20Files/Extraterm/resources/app/extraterm/src/render_process/internal/process/task_queues.js net::ERR_FILE_NOT_FOUND
retrieveSourceMapURL @ C:\Program Files\Extraterm\resources\app\extraterm\node_modules\source-map-support\source-map-support.js:81
retrieveSourceMap @ C:\Program Files\Extraterm\resources\app\extraterm\node_modules\source-map-support\source-map-support.js:110
mapSourcePosition @ C:\Program Files\Extraterm\resources\app\extraterm\node_modules\source-map-support\source-map-support.js:140
wrapCallSite @ C:\Program Files\Extraterm\resources\app\extraterm\node_modules\source-map-support\source-map-support.js:303
(anonymous) @ C:\Program Files\Extraterm\resources\app\extraterm\node_modules\source-map-support\source-map-support.js:338
prepareStackTrace @ C:\Program Files\Extraterm\resources\app\extraterm\node_modules\source-map-support\source-map-support.js:337
C:\Program Files\Extraterm\resources\app\extraterm\node_modules\source-map-support\source-map-support.js:47 GET file:///C:/Program%20Files/Extraterm/resources/app/extraterm/src/render_process/internal/process/task_queues.js net::ERR_FILE_NOT_FOUND`

**Comment by sedwards2009** [comment]
No, it is not meant to do that. Which version of Extraterm and Windows are you using?

**Comment by Kynolin** [comment]
Out of curiosity, do the native Command Prompt or PowerShell consoles do this when you run `cmd` or `powershell` from within an existing console window (not Extraterm)? For example, if I run `cmd` it just goes into a fresh prompt in the same window, but what you're describing acts more like running `start cmd` which opens command prompt in a new window.

Both CMD and PowerShell work for me with the standard executables "cmd.exe" and "powershell.exe" on Windows 10 1803 and 1903, but if you think of any settings that might attribute to this, I can try to reproduce it.

**Comment by sedwards2009** [comment]
Can I assume that you haven't changed any settings or anything?

For older versions of Windows 10 what Extraterm does (and many other terminals on Windows) is open a hidden console window in the background and then 'mirror' its contents into the terminal emulator. For some reason it *looks* like the hidden console window isn't being hidden.

More recent versions of Windows have a better and more robust way of allowing 3rd party terminal emulators which doesn't require these kinds of tricks going on.
